lib/perl/Genome/Site/TGI/Extension/Sys.pm

```diff
@@ -128,21 +126,19 @@ sub _bsub_and_wait_for_completion__validate_args {
     return(\%special_args, %bsub_args);
 }
 
-sub _bsub_and_wait_for_completion__wait_on_jobs {
-    my($class, $waiting_socket, $on_complete_cb, %seq_to_job_id) = @_;
-
-    while (%seq_to_job_id) {
-        my $fh = $waiting_socket->accept();
-        my $seq = $fh->getline;
-        $fh->close;
-        $seq =~ s/\r|\n//g;
-
-        my $job_id = delete $seq_to_job_id{$seq};
-        unless ($job_id) {
-            Carp::croak("Got unexpected response '$seq' while waiting for these jobs:\n",
-                join("\n", map { join(' => ', $_, $seq_to_job_id{$_}) } keys %seq_to_job_id));
+sub _bsub_and_wait_for_completion__wait_on_jobs_with_lsf_dependency {
+    my ( $class, $seq_to_job_id, $bsub_args, $on_complete_cb ) = @_;
+    local $ENV{LSF_NIOS_JOBSTATUS_INTERVAL} = 1;
```

Comment:
The flag is very important.  It might cause problem if it is not there.  Currently, it is here to be fail safe until we can trust the flag is always set on the blade.